### PR TITLE
Add LineageViewActionBar

### LINKED ISSUE #147: [MLMD Lineage] Implement breadcrumb bar for visualizing Lineage Explorer navigation history
**Describe the solution you'd like**

Adds an affordance below the tab bar to displays the user's navigation history in the lineage explorer.

**Anything else you would like to add:**

Mocks: https://gallery.googleplex.com/projects/MCHbtQVoQ2HCZV4FTvhm3v2g/files/MCEJu8Y2hyDScbPGl48LgCFyDIbBRSphdIw

/kind feature
/area front-end
/area metadata
/assign @kwasi 
/cc @avdaredevil

### PR DESCRIPTION
**Pull request by kwasi**
## Feature

* Adds a toolbar with breadcrumbs and a reset button to the top LineageView
* Breadcrumbs update when an Lineage card with a target is clicked
* Breadcrumbs show active, inactive, and hover states
* Client-side history navigation using reset button
* Client-side history navigation when a breadcrumb is clicked

Mocks: https://gallery.googleplex.com/projects/MCHbtQVoQ2HCZV4FTvhm3v2g/files/MCEJu8Y2hyDScc3J36YgKln9aLSFm3NIoRQ

## Implementation

Fixes #147 

Screencast
https://screencast.googleplex.com/cast/NTgxNjQ1NjMxNjk3NzE1Mnw4YTQ1MmY5ZC01MQ

Dev Deployment
https://kubeflow-ui-07e.endpoints.kwasinti-kubeflow-dev.cloud.goog/_/metadata/#/artifacts

_How fake is the data?_

This is still using fake data but using real types. `LineageView.tsx` renders the target card using the real `ml_metadata.Artifact` object loaded with `ArtifactDetails`. When a `<LineageCardRow />` with fake data is clicked, an `ml_metadata.Artifact` is created at runtime an used as the new target Artifact to breadcrumb state. Once the `LineageView` cards have access to an Artifact, they should be able to pass it up with no further changes.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/metadata/170)

### CODE REVIEW

File: frontend/src/components/LineageCardRow.tsx [hunk 1]
```diff
@@ -27,14 +37,30 @@ export class LineageCardRow extends React.Component<LineageCardRowProps> {
...
+    // TODO: Return this.props.artifact once available.
+    const mockArtifact = new Artifact();
+    const nameValue = new Value();
+    nameValue.setStringValue(this.props.title);
+    mockArtifact.getPropertiesMap().set(ArtifactProperties.NAME, nameValue);
+    return mockArtifact
+  }
+
+  private handleClick() {
+    if (this.props.setLineageViewTarget) {
```
Comment (avdaredevil) [inline frontend/src/components/LineageCardRow.tsx]: Invert to decrease code depth
```suggestion
    if (!this.props.setLineageViewTarget) return;
```

File: frontend/src/pages/LineageView.tsx [hunk 2]
```diff
@@ -74,29 +88,39 @@ class LineageView extends Page<{}, LineageViewState> {
...
-        <LineageCardColumn
-          type='execution'
-          cards={mockExec}
-          title={`${columnNames[3]}`} />
-        <LineageCardColumn
-          type='artifact'
-          cards={mockOutputArtifacts}
-          reverseBindings={true}
-          title={`${columnNames[4]}`} />
+      <div className={classes(commonCss.page)}>
```
Comment (avdaredevil) [inline frontend/src/pages/LineageView.tsx]: I see that the CSS is showing in the cluster page. But where was it moved to?

File: frontend/src/pages/LineageView.tsx [hunk 3]
```diff
@@ -125,6 +149,28 @@ class LineageView extends Page<{}, LineageViewState> {
     // });
     return '';
   }
+
+  // Updates the view and action bar when the target is set from a lineage card.
+  private setTargetFromLineageCard(target: Artifact) {
+    const actionBarRefObject = this.actionBarRef as RefObject<LineageActionBar>;
+    if (!actionBarRefObject.current) {
```
Comment (avdaredevil) [inline frontend/src/pages/LineageView.tsx]: _**nit**_ condense:
```suggestion
    if (!actionBarRefObject.current) return;
```

Comment (kwasi) [reply frontend/src/pages/LineageView.tsx]: Done

File: frontend/src/components/LineageCardRow.tsx (hunk 1 above)
Comment (kwasi) [reply frontend/src/components/LineageCardRow.tsx]: Done

File: frontend/src/pages/LineageView.tsx (hunk 2 above)
Comment (kwasi) [reply frontend/src/pages/LineageView.tsx]: Nested below the action bar with the card column container

Comment (avdaredevil) [reply frontend/src/pages/LineageView.tsx]: Gotcha, thanks!

**Pull request merged**